### DISCUSSION TITLE
Configuration for a working setup with VSCODE

### DISCUSSION THREAD

**Discussion by marcus-pousette** [libp2p/js-libp2p#2383] [General]
- **Version**:
 "version": "1.0.0",

- **Platform**:
Darwin MacBook-Pro-3.local 23.0.0 Darwin Kernel Version 23.0.0

- **Subsystem**:
Mono repo 

#### Severity:
Low

#### Description:

I am trying to debug issues I have with libp2p but find it challenging getting a working environmnet up where I can run and debug *individual* tests in VSCODE by clicking on the sidebar or on some button that appears in above test cases.   

Since this project is doing testing with Mocha I have tried 3 different extensions now but failed to use any effectively.  

Would be very happy if I could get hold of a working settings.json config together with a extension (with a specific version) that works.

I have a vague memory that I could get some test working before by not have the project open in the root, but instead open individual packages, but I can't get that to work either now.

**Comment by SgtPooki** [comment]
js-libp2p uses https://github.com/ipfs/aegir for testing, which uses playwright/mocha under the hood depending on the target environment. I usually just run tests with `npm run test:node -- -g 'test name'` to run individual tests
